File: scripts/binary/genomicranges/subtract.R

source("lib/helpers.R")
source("scripts/reading/genomicranges.R")

files <- get_binary_arguments()

annotations <- read_genomic_file(files$annotation_file)
reads <- read_genomic_file(files$read_file)


common_seqlevels <- intersect(seqlevelsInUse(reads), seqlevelsInUse(annotations))
reads <- keepSeqlevels(reads, common_seqlevels, pruning.mode = "tidy")
annotations <- keepSeqlevels(annotations, common_seqlevels, pruning.mode = "tidy")
print(reads)
print(annotations)

fragments = subtract(annotations, reads, ignore.strand = TRUE)
result     <- unlist(fragments, use.names = FALSE) 

capture.output(print(result), file = files$outfile)

File: scripts/binary/pyranges/subtract.py
import time
from lib.helpers import get_files, write_result

start = time.time()
annotations, reads = get_files("pyranges")
print(time.time() - start, "reading")

start = time.time()
df = annotations.subtract_ranges(reads, strand_behavior="ignore")
print(time.time() - start, "overlaps")

start = time.time()
write_result("binary", str(len(df)))
print(time.time() - start)
